modules/apps/frontend-js/frontend-js-jquery-web/src/main/resources/META-INF/resources/jquery/form.js

```diff
@@ -243,6 +243,14 @@
 				var oldSuccess = options.success || function () {};
 				callbacks.push(function (data) {
 					var fn = options.replaceTarget ? 'replaceWith' : 'html';
+
+					// Validate `data` through `HTML encoding` when passed `data` is passed
+					// to `html()`, as suggested in https://github.com/jquery-form/form/issues/464
+
+					fn == 'html'
+						? (data = $.parseHTML($('<div>').text(data).html()))
+						: '';
```

Comment:
Took a while to understand what this was doing and figure out you're using the ternary conditional to mutate `data` and to do that we use the previous value of `fn`... for a while it felt like you might be mutating `fn`. I think we should be able to write this in a simpler way doing something like:

```javascript
var fn = options.replaceTarget ? 'replaceWith' : 'html';

data = options.replaceTarget ? data : $.parseHTML(...);
```

This way it's obvious that `data` is being mutated and it also depends on the same condition rather than on the evaluation of the first one.

Also, I know this is the original solution and this feels like idiomatic jQuery, but is the `.parseHTML` really necessary? isn't it enough with `$('<div>').text(data).html()`?

I think this translates to the vanilla version which is:

```javascript
function sanitizeHTML(data) {
    const div = document.createElement('div');

    div.textContent = data;

    return div.innerHTML;
}
```

Reply:
I would rather keep the `.parseHTML ` to respect the original solution and simply as I am not sure if it would cause any security issue or not

Reply:
I sent a refactor commit, besides the removal of `.parseHTML` I followed up on everything. Thanks